finagle-core/src/main/scala/com/twitter/finagle/loadbalancer/RoundRobin.scala

```diff
@@ -0,0 +1,105 @@
+package com.twitter.finagle.loadbalancer
+
+import com.twitter.finagle.{ClientConnection, NoBrokersAvailableException, Service,
+  ServiceFactory, ServiceFactoryProxy, Status}
+import com.twitter.finagle.service.FailingFactory
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.finagle.util.OnReady
+import com.twitter.util.{Activity, Future, Promise, Time}
+
+/**
+ * A simple round robin balancer that chooses the next backend in
+ * the list for each request.
+ */
+class RoundRobinBalancer[Req, Rep](
+  protected val activity: Activity[Traversable[ServiceFactory[Req, Rep]]],
+  protected val statsReceiver: StatsReceiver,
+  protected val emptyException: NoBrokersAvailableException,
+  protected val maxEffort: Int = 5
+) extends ServiceFactory[Req, Rep]
+    with Balancer[Req, Rep]
+    with Updating[Req, Rep] {
+
+  // For the OnReady mixin
+  private[this] val ready = new Promise[Unit]
+  override def onReady: Future[Unit] = ready
+
+  protected[this] val maxEffortExhausted = statsReceiver.counter("max_effort_exhausted")
+
+  override def apply(conn: ClientConnection): Future[Service[Req,Rep]] = {
+    dist.pick()(conn)
+  }
+
+  protected class Node(val factory: ServiceFactory[Req, Rep])
+      extends ServiceFactoryProxy[Req,Rep](factory)
+      with NodeT[Req,Rep] { self =>
+    type This = Node
+    // We don't care about these metrics for round robin
```

Comment:
we don't use them for balancing, but they could be useful for metrics, since they're used to calculate load on a service.  maybe worth mentioning that the stats we export for "load" is not right.

https://github.com/twitter/finagle/blob/develop/finagle-core/src/main/scala/com/twitter/finagle/loadbalancer/Balancer.scala#L94-L96